htdocs/compta/facture/class/facture.class.php

```diff
@@ -1178,6 +1178,44 @@ public function createFromClone(User $user, $fromid = 0)
 			    unset($object->lines[$i]);
 			    unset($object->products[$i]); // Tant que products encore utilise
 			}
+
+			// Bloc to update dates of service (month by month only if previously filled at 1d near start or end of month)
+			// If it's a service with start and end dates
+			if ($line->product_type == 1 && !empty($line->date_start) && !empty($line->date_end) ) {
+				// Get the dates
+				$start = new DateTime();
```

Comment:
new DateTime must not be used as it does not take into account the timezone of the server, of the login user, etc...
Ude dol_mktime or dol_now to get a date and function dol_time_plus_duree to make operation on a date to get same date next month.